dev/NavigationView/NavigationViewItemsFactory.cpp

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "NavigationViewItemsFactory.h"
+#include "NavigationViewItemBase.h"
+#include "NavigationViewItem.h"
+
+NavigationViewItemsFactory::NavigationViewItemsFactory()
+{
+}
+
+winrt::UIElement NavigationViewItemsFactory::GetElementCore(winrt::ElementFactoryGetArgs const& args)
+{
+    winrt::IInspectable newContent = args.Data();
+    // Attempt to get NavigationViewItemBase from user defined IElementFactory
+    if (m_userElementFactory)
+    {
+        // Convert to args argument in the Windows::UI::Xaml namespace
+        m_getArgsWUX.Data(args.Data());
+        m_getArgsWUX.Parent(args.Parent());
+        auto element = m_userElementFactory.GetElement(m_getArgsWUX);
+
+        if (element.try_as<winrt::NavigationViewItemBase>())
+        {
+            return element;
+        }
+
+        newContent = element;
+    }
```

Comment:
It looks like if the user's element factory does not create a NavigationViewItemBase(NVIB) but the original data was a NVIB then we will ignore the users factory.  Is that intended? At face value it seems wrong to me.

suggestion:
```
auto const newContent = [userElementFactory = m_userElementFactory](){
    if(userElementFactory)
    {
        m_getArgsWUX.Data(args.Data());
        m_getArgsWUX.Parent(args.Parent());
        return m_userElementFactory.GetElement(m_getArgsWUX);        
    }
    return args.Data();
}();

if(newContent.try_as<winrt::NavigationViewItemBase>())
{
    return newContent;
}
```
This allows you to declare your local as const, removes the deeply nested return statement for the outer function, avoids setting your local to a scratch value, and doesn't bypass the userElementFactory in the situation described above